runtime/gc_glue_java/ConcurrentMarkingDelegate.cpp

```diff
@@ -171,22 +127,14 @@ MM_ConcurrentMarkingDelegate::scanThreadRoots(MM_EnvironmentBase *env)
 		}
 	}
 
-	markSchemeStackIteratorData localData;
-	localData.concurrentMarkingDelegate = this;
+	MM_MarkingDelegate::StackIteratorData localData;
+	localData.markingDelegate = _markingDelegate;
 	localData.env = env;
-	/* In a case this thread is a carrier thread, and a virtual thread is mounted, we will scan virtual thread's stack. */
-	GC_VMThreadStackSlotIterator::scanSlots(vmThread, vmThread, (void *)&localData, concurrentStackSlotIterator, true, false);
+	GC_VMThreadStackSlotIterator::scanSlots(vmThread, vmThread, (void *)&localData, stackSlotIteratorForMarkingDelegate, true, false);
 
 #if JAVA_SPEC_VERSION >= 19
 	if (NULL != vmThread->currentContinuation) {
-		GC_VMThreadStackSlotIterator::scanSlots(vmThread, vmThread, vmThread->currentContinuation, (void *)&localData, concurrentStackSlotIterator, true, false);
-#if JAVA_SPEC_VERSION >= 24
-		GC_ContinuationSlotIterator continuationSlotIterator(vmThread, vmThread->currentContinuation);
-
-		while (J9Object **slot = continuationSlotIterator.nextSlot()) {
-			doContinuationSlot(env, slot, &continuationSlotIterator);
-		}
-#endif /* JAVA_SPEC_VERSION >= 24 */
+		_markingDelegate->scanContinuationNativeSlotsNoSync(env, vmThread, vmThread->currentContinuation, true);
```

Comment:
For easier understanding what we pass as `true` you can do:
const bool stackFrameClassWalkNeeded = true; 
(not completely sure why we pass true, but if you know you can provide a comment)